src/runners.py

```diff
@@ -218,16 +219,74 @@ def get_datalog_files(self) -> List[str]:
     def decomp_out_produced(self, out_dir: str) -> bool:
         pass
 
+    @abstractmethod
+    def match_pattern(self, contract_filename: str) -> bool:
+        pass
+
+
+class MixedFactGenerator(AbstractFactGenerator):
+
```

Comment:
Missing type annotations for the fields and some function parameters in this class.